### PR TITLE
Remove Java versions to CI matrix.

### PR DESCRIPTION
**Pull request by dperez37**
This PR updates the `ci.yml` file to use fewer Java versions and updates the versions of the GitHub Actions we use.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -54,8 +47,9 @@ jobs:
           bundler-cache: true
 
       - name: Configure JDK
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
+          distribution: 'zulu'
```
Comment (calebjderosier) [inline .github/workflows/ci.yml:52]: We always want it to be `zulu`, even if we aren't sure what machine it's running on?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,19 +15,12 @@ jobs:
       fail-fast: false
       matrix:
         java-version:
-          - 1.8
-          - 9
-          - 10
+          - 8
```
Comment (calebjderosier) [inline .github/workflows/ci.yml:18]: Do you know what the difference is here between `8` and `1.8` as far as this change is concerned?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -15,19 +15,12 @@ jobs:
...
           - 11
-          - 12
-          - 13
-          - 14
-          - 15
-          # - 16 TODO: Include JDK 16 when kapt supports it: https://youtrack.jetbrains.com/issue/KT-45545
```
Comment (calebjderosier) [inline .github/workflows/ci.yml:26]: Just no longer interested in 16?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dperez37) [reply .github/workflows/ci.yml:52]: This is actually the only distribution supported before, in the new version of the action, they added support for more and I just had to specify the one we used before.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dperez37) [reply .github/workflows/ci.yml:18]: The new version of the action does not support the old term of `1.8`, they want `8` now. Same thing though.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dperez37) [reply .github/workflows/ci.yml:26]: Naw, just the main 2 we use in Android

File: .github/workflows/ci.yml (hunk 1 above)
Comment (calebjderosier) [reply .github/workflows/ci.yml:52]: Oh fair enough, sounds good!

**Review by calebjderosier — APPROVED** [review]

**Comment by dperez37** [comment]
Thanks @calebjderosier

**Pull request merged**